drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scenariosimulation/backend/server/ScenarioSimulationServiceImpl.java

```diff
@@ -347,15 +347,15 @@ protected void createActivatorIfNotExist(Path context) {
         Package junitActivatorPackage = getOrCreateJunitActivatorPackage(kieModule);
         final org.uberfire.java.nio.file.Path activatorPath = getActivatorPath(junitActivatorPackage);
 
-        if (!ioService.exists(activatorPath)) {
+        boolean removeExistingActivator = ensureDependencies(kieModule);
+
+        if (removeExistingActivator || !ioService.exists(activatorPath)) {
+            removeOldActivatorIfExists(activatorPath, kieModule);
```

Comment:
I am missing something, method says if exists `removeOldActivatorIfExists` but the if that wrap, says if it doesn't exists `!ioService.exists(activatorPath)`, to me seems like contradiction.

Reply:
I added some comments to clarify. There are two reasons why the activator needs to be saved, if it doesn't exist or if it needs to be migrated (I changed `removeExistingActivator` variable to `needMigrateActivatorIfExists`)

Reply:
@danielezonca sorry, my mind is blank today, I can not understand this. In `if` we had statement `!ioService.exists(activatorPath)` that suggests me without any deep investigating, it is evaluated to `true` if `activatorPath`  does not exist and then `removeOldActivatorIfExists(activatorPath, kieModule)` tries to remove something at the same path, that we evaluated previously as it does not exist? I think I am missing something.

Reply:
The method `removeOldActivatorIfExists` looks for existing activators in multiple paths, the idea is to be sure that after that method no existing activators are in the project.
`!ioService.exists(activatorPath)` verify only if an activator exists in the target position